spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -82,13 +94,20 @@ public TwoStepsConversions(CustomConversions customConversions) {
 
 	@Override
 	@SuppressWarnings("unchecked")
-	public <T> T convertOnRead(Value val, Class<?> targetType, Class<?> componentType) {
+	public <T> T convertOnRead(Value val, DatastorePersistentProperty persistentProperty) {
+		BiFunction<Object, Class<?>, T> readConverter = persistentProperty.isEmbedded()
+				? (value, targetClass) -> (T) this.datastoreEntityConverter.read(targetClass, (BaseEntity) value)
+				: this::convertOnReadSingle;
+
+		Class<?> targetType = persistentProperty.getType();
+
+		Class<?> componentType = persistentProperty.getComponentType();
```

Comment:
If the user had not included the inner type in his POJO definition, what would the error here look like? 

For example if he had used `List myNumbers` instead of `List<Double> myNumbers`. Should we validate this early on when creating the DatastorePersistentEntity and throw an exception telling the user this error?